Mathlib/Algebra/MonoidAlgebra/NoZeroDivisors.lean

```diff
@@ -46,119 +50,60 @@ namespace AddMonoidAlgebra
 
 open Finsupp
 
+instance {L σ : Type*} [LinearOrder σ] [LinearOrder L] [AddGroup L]
```

Comment:
Is `LinearOrder σ` really needed as an assumption, or can you get away with adding an arbitrary linear order inside the proof, like in the final instance below.  (I.e. extend the Partial order given by `(• = •)` to a linear one inside the proof.)

Reply:
The way to put an arbitrary linear order on a type is `IsWellOrder.linearOrder WellOrderingRel`. However I'm gonna prove this from the `Pi` result assuming only `UniqueSums L`.

Reply:
Ok, I'm happy to see orders slowly fading away!

Reply:
Anyway, while waiting for the `Pi` proof, I remove the `LinearOrder σ` assumption in this instance.